VMs/metrics/6_start_clone_agent.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+# Copyright (c) 2018 Intel Corporation
+# 
+# SPDX-License-Identifier: Apache-2.0
+
+JARSRC="${HOME}/bin/agent.jar"
+JARDEST_DIR="/var/lib/jenkins"
+JARDEST="${JARDEST_DIR}/agent.jar"
+
+LOGFILE=/dev/null
+
+source global_env.sh
+
+die() {
+	echo "Fatal: $*" >&2
+	exit 1
+}
+
+error() {
+	echo "Error: $*" >&2
+}
+
+message() {
+	echo "Message: $*"
+}
+
+main()
+{
+	if [ ! -f ${CONFIG_DIR}/${ENV_FILE} ]; then
+		die "Env file ${CONFIG_DIR}/${ENV_FILE} missing"
+	fi
+
+	source ${CONFIG_DIR}/${ENV_FILE}
+
+	# Ensure we do not have an existing instance
+	message "Ensuring ${CONFIG_DIR} is shut down"
+	./7_delete_clone.sh ${CONFIG_DIR}
+
+	# And make the clone
+	message "Cloning ${VM_NAME} -> ${VM_CLONE_NAME}"
+	./4_clone_vm.sh ${CONFIG_DIR}
+
+	# And kick it off
+	message "Starting clone VM $VM_CLONE_NAME"
+	virsh start ${VM_CLONE_NAME}
+
+	message "Napping for ${VM_BOOTTIME}s for VM to come up"
+	sleep ${VM_BOOTTIME}
+
+	VM_IP=$(virsh domifaddr ${VM_CLONE_NAME} | tail -2 | head -1 | awk '{print $4}')
+	VM_IP=${VM_IP%%/*}
+
+	echo "IP of ${VM_CLONE_NAME} is ${VM_IP}"
+
+	ssh -F /dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o IdentitiesOnly=yes -i ${PRIVKEYNAME} ${VM_USERNAME}@${VM_IP} sudo mkdir -p ${JARDEST_DIR}
+	ssh -F /dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o IdentitiesOnly=yes -i ${PRIVKEYNAME} ${VM_USERNAME}@${VM_IP} sudo chown jenkins ${JARDEST_DIR}
+	scp -F /dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o IdentitiesOnly=yes -i ${PRIVKEYNAME} ${JARSRC} ${VM_USERNAME}@${VM_IP}:${JARDEST}
+}
+
+if [ $# != 1 ]; then
+	error "Missing first argument"
+	message "Usage: $0 distro_config_dirname"
+	die
```

Comment:
This would print a `Fatal:` without any other message?  this may create confusion

Reply:
I've found a few more dangling 'die's as well - will have a sweep